src/main/java/com/labsynch/labseer/chemclasses/bbchem/BBChemStructureServiceImpl.java

```diff
@@ -194,23 +304,66 @@ public BBChemParentStructure getProcessedStructure(String molfile, Boolean inclu
 				JsonNode molecularFormulaNode = responseJsonNode.get("molecular_formula");
 				bbChemStructure.setMolecularFormula(molecularFormulaNode.asText());
 
-				if(includeFingerprints) {
-					JsonNode fingerprintNode = responseJsonNode.get("fingerprint");
-					bbChemStructure.setSubstructure(getFingerprint(bbChemStructure.getMol(), "substructure_search"));
-					bbChemStructure.setSimilarity(getFingerprint(bbChemStructure.getMol(), "similarity_score"));
-				}
+				// Set recorded date todays date. This simplifies the code when we need to persist the structure
+				// in various places.
+				bbChemStructure.setRecordedDate(new Date());
+
+				//Get get hasmap name of structure at index i
+				String structureKey = structuresArray[i].toString();
+				
+				// Add to the map
+				processedStructures.put(structureKey, bbChemStructure);
 			}
 
+			if(includeFingerprints) {
+				// Input fingerprint hashes
+				HashMap<String, String> processedStructureHash = new HashMap<String, String>();
+				for (String structureId : processedStructures.keySet()){
+					processedStructureHash.put(structureId, processedStructures.get(structureId).getMol());
+				}
+
+				// Get the fingerprints
+				HashMap<String, BitSet> substructureHashMap = getFingerPrints(processedStructureHash, "substructure_search");
+				HashMap<String, BitSet> similarityHashMap = getFingerPrints(processedStructureHash, "similarity_score");
+
+				// Add the substructure fingerprints to the processed structures
+				for (String structureId : substructureHashMap.keySet()){
+					processedStructures.get(structureId).setSubstructure(substructureHashMap.get(structureId));
+				}
+
+				// Add the similarity fingerprints to the processed structures
+				for (String structureId : similarityHashMap.keySet()){
+					processedStructures.get(structureId).setSimilarity(similarityHashMap.get(structureId));
+				}
+			}
 
+			return processedStructures;
 		} catch (CmpdRegMolFormatException e) {
-			logger.error("Error posting to preprocessor service: " + e.getMessage());
+			logger.error("Error processing structures: " + e.getMessage());
 			throw new CmpdRegMolFormatException(e);
 		}
 
-		// Set recorded date todays date. This simplifies the code when we need to persist the structure
-		// in various places.
-		bbChemStructure.setRecordedDate(new Date());
-		return bbChemStructure;
+	}
+
+	@Override
+	public BBChemParentStructure getProcessedStructure(String molfile, Boolean includeFingerprints) throws CmpdRegMolFormatException {
+		BBChemParentStructure bbChemStructure = new BBChemParentStructure();
+		// Post to the service and parse the response
+		try {
+			// Input hashmap 
+			HashMap<String, String> structures = new HashMap<String, String>();
+			structures.put("molstructure", molfile);
```

Comment:
I think I understand.  You were confused because its not obvious that molStructure doesn't really have a meaning and is really just a made up key name?   Would something like `TmpMolKey01` be more obvious?

Reply:
Just added that with some code comments